pages/demystify_dsl.md

```diff
@@ -0,0 +1,310 @@
+# Demystify Ewebmachine DSL
+
+Everyone has made a route with `Ewebmachine` (or at least copy and paste one),
+but did you wonder once how does it work under the hood. Maybe you start
+looking into it and were repelled by the heavy use of macro.
+
+This document aims to go through some of `Ewebmachine` internals, to explain
+how from a bunch of macros, we end up with a Plug pipeline.
+
+---
+
+Let's start with this small module:
+```elixir
+defmodule MyApi do
+  use Ewebmachine.Builder.Resources
+
+  resource "/api/path" do after
+     allowed_methods do: ["GET"]
+
+     defh(to_html, do: "<h1>HTML</h1>")
+  end
+end
+```
+
+It imports the macro `Ewebmachine.Builder.Resources.resource/[3-4]` into the
+scope, with which we can make the `/api/path` route.
+
+From this point on, the macro's magic starts :).
+
+**How do handlers (`allowed_methods` and friends) work?**
```

Comment:
Those bold titles could actually be titles. Would be better for this file's structure !
```suggestion
## How do handlers (`allowed_methods` and friends) work?
```